test/sqllogictest/joins.slt

```diff
@@ -602,3 +602,92 @@ CREATE TABLE join_fail (la date);
 
 query error NATURAL/USING join column "la" cannot be cast to uniform type: integer vs date
 SELECT la FROM l JOIN join_fail USING (la)
+
+# test that joins properly handle null keys
```

Comment:
For this query, the plan is flipped on its head, that is, the relations are read in opposite order and the Filters being applied are also flipped. As a result, this branch and main return completely diverging results:

```
SELECT *  FROM t1  AS a1  RIGHT  JOIN t2  AS a2  ON ( a1 .f2  = a2 .f1  )  WHERE a1 .f1  IS  NULL   ;
```

Reply:
This has also been fixed in the latest version of the PR branch.